pkg/imageset/generate.go

```diff
@@ -185,7 +185,7 @@ func GenerateImageSetJob(cd *hivev1.ClusterDeployment, imageSet *hivev1.ClusterI
 
 // GetImageSetJobName returns the expected name of the imageset job for a ClusterImageSet.
 func GetImageSetJobName(cdName string) string {
-	return fmt.Sprintf("%s-imageset", cdName)
+	return apihelpers.GetResourceName(cdName, "imageset")
```

Comment:
Not for this PR but I think we should establish a convention that any dependent object we create, but don't want to explicitly add as a link in the API, must have a hive.openshift.io/cluster-deployment-name label.